src/constraint_system/scalar_mul.rs

```diff
@@ -0,0 +1,236 @@
+        }
+
+        // Depending on the round, or we call the different gates,
+        // from the composer and push the wires to accordingly in 
+        // the programme memory.
+        if i.0 == 0 {
+            // TODO: remove Fr unwraps
+            composer.fixed_base_add_with_initial(
+                Fr::from_bytes(&AffinePoint::from(mul_points[i.0]).get_x().to_bytes()).unwrap(),
+                Fr::from_bytes(&AffinePoint::from(mul_points[i.0]).get_y().to_bytes()).unwrap(),
+                x_alpha,
+                accums[i.0],
+            );
+        } else if i.0 < wnaf_scalar.len() - 1 {
+            composer.fixed_base_add(
+                Fr::from_bytes(&AffinePoint::from(mul_points[i.0]).get_x().to_bytes()).unwrap(),
+                Fr::from_bytes(&AffinePoint::from(mul_points[i.0]).get_y().to_bytes()).unwrap(),
+                x_alpha,
+                accums[i.0],
+                256,
+            );
+        } else {
+            let var_a = composer.add_jubjub_input(
+                Fr::from_bytes(&AffinePoint::from(mul_points[i.0]).get_x().to_bytes()).unwrap(),
+            );
+            let var_b = composer.add_jubjub_input(
+                Fr::from_bytes(&AffinePoint::from(mul_points[i.0]).get_y().to_bytes()).unwrap(),
+            );
+            let var_c = composer.add_jubjub_input(x_alpha);
+            let var_d = composer.add_jubjub_input(accums[i.0]);
+            composer.big_add_gate(
+                var_a,
+                var_b,
+                var_c,
+                var_d,
+                Scalar::zero(),
+                Scalar::zero(),
+                Scalar::zero(),
+                Scalar::zero(),
+                Scalar::zero(),
+                Scalar::zero(),
+            );
+        }
+    }
+
+    composer.w_o[composer.w_o.len() - 1]
+}
+
+// Now that we have our scalar in the NAF form, we need to precompute our look up table.
+// This is done to provide the potential add-in-x-coordinate's at each round. As only 1
+// and 3 are possible, the ladder look up function will be a 2 bit output. The 2 resulting
+// output coordinates, along with their negative y-coordinate counterparts, will be the
+// 4 coordinates, which the add-in-x-coordinate is derived from.
+
+#[test]
+fn test_ladder() {
+    let ladder = fixed_base_ladder();
+    assert_eq!(ladder[0], GENERATOR);
+    assert_eq!(
+        ladder[1],
+        AffinePoint::from(ExtendedPoint::from(GENERATOR) * Fr::from(3 as u64))
+    );
+}
+
+#[test]
+fn test_scalar_mul() {
+    let mut composer = StandardComposer::default();
+    let point = GENERATOR;
+    let gen_x = point.get_x();
+    let gen_y = point.get_y();
+    let scalar = Fr::from(31 as u64);
+
+    // TODO: what do we need to check?
+    // assert_eq!(scalar_mul(scalar, &mut composer), 6749237362536748595030987654329305826145364798328171542537485058472526649593, 435472901526383203736325467483956216236323421376357876155393487519030193845 )
+}
```

Comment:
This is not even testing anything.

The test that we asked for should compute inside and outside the circuit a  scalar mul and check that the results are the same.